src/Routes/Routes.tsx

```diff
@@ -170,51 +172,69 @@ function useLocationBackground() {
   return { background, location }
 }
 
-export const Routes = (props: { additionalRoutes?: Array<NestedRoute> }) => {
+export interface IRouteContext {
+  appRoutes: NestedRoute[]
+  currentRoute: NestedRoute | void
+}
+
+const RouteContext = createContext<IRouteContext | null>(null)
+
+export const AppRouteProvider: React.FC = ({ children }) => {
+  const location = useLocation()
+  const { routes: pluginRoutes } = usePlugins()
+
+  const appRoutes = useMemo(() => {
+    return generateAppRoutes(union(pluginRoutes, routes))
+  }, [pluginRoutes])
+
+  const currentRoute = useMemo(() => {
+    return appRoutes.find(e => matchPath(location.pathname, { path: e.path, exact: true }))
+  }, [appRoutes, location.pathname])
+
+  return (
+    <RouteContext.Provider value={{ currentRoute, appRoutes }}>{children}</RouteContext.Provider>
+  )
+}
+
+export const useAppRoutes = (): IRouteContext =>
```

Comment:
this is dangerous - you're casting something possibly undefined to a concrete type

https://github.com/shapeshift/web/blob/develop/src/context/PluginProvider/PluginProvider.tsx#L135 is how to implement this safely

Reply:
otherwise i will gladly give big boy stamp and 🍭